src/common/TargetPath/targetPath.util.js

```diff
@@ -436,4 +437,32 @@ export const getFeatureVector = (dispatch, project, projectItem, setDataInputSta
     .catch(error => {
       showErrorNotification(dispatch, error, '', 'Failed to fetch feature vector data')
     })
-}
+}, 300)
+
+export const prepareTargetPathInitialState = (inputDefaultState, inputDefaultValue = '', selectDefaultValue = '') => {
+  if (inputDefaultState) {
+    return inputDefaultState
+  } 
+  
+  if (inputDefaultValue && selectDefaultValue.startsWith('store')) {
+    const state = {...targetPathInitialState}
+    const { key, project, kind } = parseUri(selectDefaultValue + inputDefaultValue)
+    const projectItemReference = inputDefaultValue.split(`${kind}/${project}/${key}`)?.[1] || ''
+    state.storePathType = kind
+    state.project = project
+    state.projectItem = key
+    state.inputProjectItemPathEntered = true
+    state.inputProjectPathEntered = true
+    state.inputStorePathTypeEntered = true
+
+    
+    if (projectItemReference) {
+      state.projectItemReference = projectItemReference
+          state.inputProjectItemReferencePathEntered = true
+    }
+
+    return state
+  }
+  
```

Comment:
- delete 1 empty line
- format the code (`state.inputProjectItemReferencePathEntered = true` has redundant space)